### PR TITLE
Add create, update, retrieve, and delete to Meter API

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?

Meters are only importable via files (ESPM or Green Button). This prevented the ability to delete or update meters.

#### What's this PR do?
* update version of pre-commit autoflake8
* nest meter API under properties
* Add "manual entry" type of meter source, which is not used yet.
* Create meter serializer to handle CRUD
* Add "alias" to meter fields

#### How should this be manually 

Most of the functionality for this PR is handled through the API. Ideally, using postman (or similar) try:
* Creating new meter
* Deleting meter
* Updating

#### What are the relevant tickets?
#3638 
#3640

### CODE REVIEW

File: seed/data_importer/utils.py [hunk 1]
```diff
@@ -408,9 +408,12 @@ def usage_point_id(raw_source_id):
     """
     Extracts and returns the usage point ID of a GreenButton full uri ID.
     """
-    id_split = raw_source_id.split('/')
-    usage_point_index = next(i for i, substr in enumerate(id_split) if substr == "UsagePoint") + 1
-    return id_split[usage_point_index]
+    if raw_source_id and '/' in raw_source_id:
```
Comment (nllong) [inline seed/data_importer/utils.py:411]: only split if it exists...

File: seed/tests/test_meter_views.py [hunk 2]
```diff
@@ -24,15 +28,120 @@ def setUp(self):
...
+
+        # faker class for properties
+        self.property_view_factory = FakePropertyViewFactory(organization=self.org)
+
+    def test_create_meter(self):
+        property_view = self.property_view_factory.get_property_view()
+        url = reverse('api:v3:property-meters-list', kwargs={'property_pk': property_view.id})
+
+        payload = {
+            'property_id': property_view.property.id,
```
Comment (nllong) [inline seed/tests/test_meter_views.py:63]: I'm pretty sure this is correct, but a second set of eyes will be nice! 

The thought is that we find the "property" using the property_view_id that is in the URL (which is the `reverse` command in line 60, then the meter is attached to the "property" ID which is found through the property_view.

File: seed/tests/test_property_views.py [hunk 3]
```diff
@@ -1522,8 +1522,7 @@ def test_property_meters_endpoint_returns_a_list_of_meters_of_a_view(self):
         }
         gb_gas_meter = Meter.objects.create(**meter_details)
 
-        url = reverse('api:v3:properties-meters', kwargs={'pk': self.property_view_1.id})
-        url += f'?organization_id={self.org.pk}'
+        url = reverse('api:v3:property-meters-list', kwargs={'property_pk': self.property_view_1.id})
```
Comment (nllong) [inline seed/tests/test_property_views.py:1525]: existing API is now under `property-meters-list` and does not require the passing of the org_id.

File: seed/serializers/meters.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        model = Meter
+        exclude = ('property', 'scenario',)
+
+    def to_representation(self, obj):
+        result = super().to_representation(obj)
+
+        if obj.source == Meter.GREENBUTTON:
+            result['source_id'] = usage_point_id(obj.source_id)
+
+        result['scenario_name'] = obj.scenario.name if obj.scenario else None
```
Comment (haneslinger) [inline seed/serializers/meters.py:34]: I think you want this inverted, I think it would error rn if `obj.scenario is None`.
```suggestion
        result['scenario_name'] = None if not obj.scenario else obj.scenario.name
```

File: seed/tests/test_meter_views.py [hunk 5]
```diff
@@ -2,17 +2,23 @@
...
+    kbtu_thermal_conversion_factors as conversion_factors
 from seed.landing.models import SEEDUser as User
-from seed.tests.util import DataMappingBaseTestCase
+from seed.test_helpers.fake import FakePropertyViewFactory
+from seed.tests.util import DeleteModelsTestCase
 from seed.utils.organizations import create_organization
 
 
-class TestMeterValidTypesUnits(DataMappingBaseTestCase):
+class TestMeterValidTypesUnits(DeleteModelsTestCase):
```
Comment (haneslinger) [inline seed/tests/test_meter_views.py:18]: Why this change?

File: seed/tests/test_meter_views.py (hunk 2 above)
Comment (haneslinger) [reply seed/tests/test_meter_views.py:67]: I'm a little confused. The request is `POST /api/v3/property_meters_list/<property_view.id>`, then in the body we give it the `property_id`. But like, why would the body need to specify the property, as it is a given because of the `property_view.id` in the url? 


The request being made here looks right, the actual endpoint seems a bit repetitive.

**Review by haneslinger — CHANGES_REQUESTED** [review]

**Comment by haneslinger** [comment]
Just small things.

**Comment by nllong** [comment]
These unit tests are killing me. Something non-deterministic is happening that I'm still trying to debug.

File: seed/tests/test_meter_views.py (hunk 2 above)
Comment (nllong) [reply seed/tests/test_meter_views.py]: I updated this so that the meter's viewset `perform_create` method grabs the `property_pk`. The body had the property because that was legacy, it has now been removed.

File: seed/serializers/meters.py (hunk 4 above)
Comment (nllong) [reply seed/serializers/meters.py:33]: hmm, either works. We have lots of tests where `obj.scenario` is None and it works as expected.

File: seed/tests/test_meter_views.py (hunk 5 above)
Comment (nllong) [reply seed/tests/test_meter_views.py:18]: `DataMappingBaseTestCase` has a bunch of helper methods for running the mapping and merging. It inherits from `DeleteModelsTestCase` which simply makes sure that all the models are empty when the test suite runs.

**Review by haneslinger — APPROVED** [review]
🎊
